### PR TITLE
Create garden tour

### PR DESCRIPTION
**Pull request by jamiander**
Created custom garden tour function.  Add/delete gardens from tour.  Styled garden list page and fixed centering issues.  Added link to website to convert address to lon/lat.

### CODE REVIEW

File: src/main/resources/templates/gardens/view.html [hunk 1]
```diff
@@ -29,7 +31,8 @@ <h4>Owner </h4>
         </div>
     </div>
     <div class = "row  justify-content-center">
-        <a th:href="@{/user/gardens/list}">Return to Water Gardens</a>
+        <a class="btn btn-primary" th:href="@{/user/gardens/list}">Return to Water Gardens</a>
+        <a class="btn btn-success" th:href="@{'/user/gardens/tour'}">Return to My Water Garden Tour</a>
     </div>
 </div>
```
Comment (MulesKC) [inline src/main/resources/templates/gardens/view.html:38]: did you want a footer on this page?

File: src/main/resources/templates/gardens/view.html [hunk 2]
```diff
@@ -29,7 +31,8 @@ <h4>Owner </h4>
         </div>
     </div>
     <div class = "row  justify-content-center">
-        <a th:href="@{/user/gardens/list}">Return to Water Gardens</a>
+        <a class="btn btn-primary" th:href="@{/user/gardens/list}">Return to Water Gardens</a>
```
Comment (MulesKC) [inline src/main/resources/templates/gardens/view.html]: I thought these buttons looked a little tight side-by-side, what are your thoughts on adding style="margin-right:10px;" to this first button to give them some space?

File: src/main/resources/templates/gardens/view.html [hunk 3]
```diff
@@ -29,7 +31,8 @@ <h4>Owner </h4>
         </div>
     </div>
     <div class = "row  justify-content-center">
```
Comment (MulesKC) [inline src/main/resources/templates/gardens/view.html]: if you do want a footer, when I was testing things I had to add style="margin-bottom:20px;" to this line to give some space for the footer

File: src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    public String addTourGarden(Model model, @PathVariable("gardenId") Integer gardenId) {
+
+        Optional<Garden> chosenGarden = gardenRepository.findById(gardenId);
+        if (chosenGarden.isPresent()) {
+            Garden tourGarden = chosenGarden.get();
+            if (!tourGardens.contains(tourGarden)) {
+                tourGardens.add(tourGarden);
+            }
+        }
+        model.addAttribute("gardens", gardenRepository.findAll());
```
Comment (MulesKC) [inline src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java:48]: I needed to add the following here to get back to the garden list after adding a garden that had been searched for:
        List<Feature> selectedFeatures = new ArrayList<>();

        model.addAttribute("selectedFeatures", selectedFeatures);
        model.addAttribute("features", featureRepository.findAll());;
        model.addAttribute("searchTerm", "");

**Comment by MulesKC** [comment]
I like the tour feature! I added a few comments about styling that are not super important. I did add an important comment to the TourController that fixes a small bug I think I found with adding gardens that had been searched for and getting back to the garden list. Let me know what you think!

File: src/main/resources/templates/gardens/view.html (hunk 2 above)
Comment (jamiander) [reply src/main/resources/templates/gardens/view.html]: Yeah, there’s definitely room for more styling in several areas.

File: src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        model.addAttribute("tourGardens", tourGardens);
+        model.addAttribute("title", "My Water Garden Tour");
+        return "/gardens/list";
+    }
+
+    @GetMapping("/gardens/tour")
+    public String listGardenTour(Model model) {
```
Comment (KBeas21) [inline src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java:58]: What is this endpoint doing? Is this endpoint just listing your current tours you have?

File: src/main/resources/static/styles.css [hunk 6]
```diff
@@ -34,6 +34,70 @@
     margin: 2rem auto auto auto;
 }
 
+.garden-title {
+    text-align: center;
+    margin-top: 3rem;
```
Comment (KBeas21) [inline src/main/resources/static/styles.css:39]: like the use of rems here over px good work.

Comment (KBeas21) [reply src/main/resources/static/styles.css:39]: here is a link to the difference of em, rem, px, etc: https://webdesign.tutsplus.com/tutorials/comprehensive-guide-when-to-use-em-vs-rem--cms-23984

**Review by KBeas21 — APPROVED** [review]
Great work! I would just address others' comments (if you have not already) and then send it into master that way you can branch off of it and start working on the persistence part of this if you would like.

File: src/main/resources/templates/gardens/view.html (hunk 3 above)
Comment (jamiander) [reply src/main/resources/templates/gardens/view.html]: Ok, I'll try that.  I must have deleted that by accident?  Not sure, but I'll add it in.

File: src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java (hunk 5 above)
Comment (jamiander) [reply src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java:58]: If I understand the section you're asking about, it is connected to the button at the top of Garden List that takes you to My Water Garden Tour.

File: src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java (hunk 4 above)
Comment (jamiander) [reply src/main/java/org/launchcode/water_garden_tour/controllers/TourController.java:48]: Cool, I added it.  Didn't notice that your search features had disappeared.  Good catch!

File: src/main/resources/templates/gardens/view.html (hunk 2 above)
Comment (jamiander) [reply src/main/resources/templates/gardens/view.html]: I added some margins with several other buttons, too.

File: src/main/resources/static/styles.css [hunk 7]
```diff
@@ -34,6 +34,70 @@
...
+}
+
+.card {
+    border: none;
+    margin-top: 3rem;
+    margin-bottom: 3rem;
+}
+
+.card-img-top {
+    height: 20vw;
```
Comment (hstrodtman) [inline src/main/resources/static/styles.css:69]: Looks great, and functioned well for me! My only suggestion is the photos are still being cropped, but it you remove line 69 that should fix the issue.

**Review by hstrodtman — APPROVED** [review]

**Pull request merged**